### PR TITLE
chore: use m1 machines in ci

### PR DESCRIPTION
**Pull request by brainbicycle**
This PR resolves [PHIRE-251] 

### Description

Circleci is deprecating non-m1 machines: https://discuss.circleci.com/t/macos-intel-support-deprecation-in-january-2024/48718
Plus they claim big speed ups!

Solid speed up, down to around 30 min from 40 min:
https://app.circleci.com/pipelines/github/artsy/eigen/45598/workflows/47299831-fc92-4fb3-bdab-5911bc822c17

### PR Checklist

- [ ] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- use m1 machines in ci - brian 

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[PHIRE-251]: https://artsyproduct.atlassian.net/browse/PHIRE-251?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Dev changes (use m1 machines in ci - brian)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 0c48dbe63efaba704b5daef3642b2d118e789e0f
</p>

File: .circleci/config.yml [hunk 1]
```diff
@@ -4,7 +4,6 @@ orbs:
   node: circleci/node@5.1.0
   queue: eddiewebb/queue@2.2.1
   horizon: artsy/release@0.0.1
-  ruby: circleci/ruby@2.1.0
```
Comment (brainbicycle) [inline .circleci/config.yml:7]: ran into an issue with this orb on m1 machines: https://github.com/CircleCI-Public/ruby-orb/issues/39
the machine we are using comes with 2.7.8 pre-installed which should be close enough 🤷

Comment (brainbicycle) [reply .circleci/config.yml:7]: opened a new one here since that was closed: https://github.com/CircleCI-Public/ruby-orb/issues/134

**Review by gkartalis — APPROVED** [review]
🔥 Thats awesome!

**Pull request merged**